### ISSUE TITLE
Using logical lines for <num>j and <num>k

### ISSUE THREAD

**Issue by zhiwei-nu** [syl20bnr/spacemacs#8720]
I use `evil-next-visual-line` for normal navigations, but sometimes i do `5j` to move up 5 lines, for example but it only counts visual lines. 

I have mapped the relevant keys as below and it works for operators (like `d5j`) but not for motions
```
  (define-key evil-motion-state-map "j" 'evil-next-line)
  (define-key evil-motion-state-map "k" 'evil-previous-line)
  (define-key evil-operator-state-map "j" 'evil-next-line)
  (define-key evil-operator-state-map "k" 'evil-previous-line)
  (define-key evil-normal-state-map "j" 'evil-next-visual-line)
  (define-key evil-normal-state-map "k" 'evil-previous-visual-line)
```

Is there any way to get this to work? Thanks in advance!

**Comment by CeleritasCelery** [comment]
should report upstream at [emacs-evil/evil](https://github.com/emacs-evil/evil). If they have a solution report it back here.

**Comment by zhiwei-nu** [comment]
ok, will do!

**Comment by zhiwei-nu** [comment]
for anyone else wondering how to do this:

```
(evil-define-motion my-evil-next-line (count)
    (interactive "P")
    (let ((command (if count 'evil-next-line 'evil-next-visual-line)))
      (funcall command (prefix-numeric-value count))))

  (define-key evil-motion-state-map (kbd "j") 'my-evil-next-line)
  (evil-define-motion my-evil-previous-line (count)
    (interactive "P")
    (let ((command (if count 'evil-previous-line 'evil-previous-visual-line)))
      (funcall command (prefix-numeric-value count))))

```

**Comment by CeleritasCelery** [comment]
Does this solution work any better?

```Lisp
(defun my/evil-next-line (orig-fun &rest args)
  "check to see if we are in visual line mode"
  (if visual-line-mode
	  (apply 'evil-next-visual-line args)
	(apply orig-fun args)))
(advice-add 'evil-next-line :around 'my/evil-next-line)

(defun my/evil-previous-line (orig-fun &rest args)
  "check to see if we are in visual line mode"
  (if visual-line-mode
	  (apply 'evil-previous-visual-line args)
	(apply orig-fun args)))
(advice-add 'evil-previous-line :around 'my/evil-previous-line)
```

**Comment by zweicoder** [comment]
@CeleritasCelery Thanks for still following up! I tried that one but it doesn't really work as expected. 

I have since found the solution and posted over [emacs-evil](https://github.com/emacs-evil/evil/issues/817#event-1060640478) as well.

The working config is this:
```
  (evil-define-motion my-evil-next-line (count)
    :type line
    (let ((command (if count 'evil-next-line 'evil-next-visual-line)))
      (funcall command (prefix-numeric-value count))))

(define-key evil-motion-state-map (kbd "j") 'my-evil-next-line)
```

This basically switches between `evil-next-line` and `evil-next-visual-line` depending on whether the motion was prefixed with a number
